bcda/web/api.go

```diff
@@ -305,12 +305,26 @@ func validateRequest(r *http.Request) ([]string, *fhirmodels.OperationOutcome) {
 	params, ok = r.URL.Query()["_since"]
 	if ok {
 		// Regex pattern for FHIR Instant format found at https://www.hl7.org/fhir/datatypes.html#primitive
-		instantRegex := `([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])T([01][0-9]|2[0-3]):[0-5][0-9]:([0-5][0-9]|60)(\.[0-9]+)?(Z|(\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00))`
-		validSince, err := regexp.MatchString(instantRegex, params[0])
-		if err != nil || !validSince {
+		instantRegex := `^([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])T([01][0-9]|2[0-3]):[0-5][0-9]:([0-5][0-9]|60)(\.[0-9]+)?(Z|(\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00))$`
```

Comment:
Does this change also address https://jira.cms.gov/browse/BCDA-3292 ?

Reply:
Yes

Reply:
Since the FHIR#instant type appears to the same as RFC3339 format, can we remove the regex check and instead rely on the error being returned by the `time.Parse` call?

Reply:
Unfortunately, it is NOT the same as RFC3339.  FHIR Instant allows more than 10 decimal places for "seconds", RFC3339 only allows 9. :(

Reply:
Ah darn! Picosecond resolution is pretty impressive :)

Inside the doc, it does list the following description:

```

An instant in time in the format YYYY-MM-DDThh:mm:ss.sss+zz:zz (e.g. 2015-02-07T13:28:17.239+02:00 or 2017-01-01T00:00:00Z).
--
```

Should the documentation (and regex) be updated to reflect the format only goes up to milliseconds?

Reply:
The regex for fhir instant was taken straight from the spec: https://www.hl7.org/fhir/datatypes.html#primitive
I think we should support that exactly as-is (which, unfortunately, is even more precise than RFC3339)

Reply:
Gotcha. It also matches the spec for [xsd:datetime](https://www.w3.org/TR/xmlschema-2/#dateTime)

Reply:
Did we decide on removing this regex and instead using `RFCNano3339` ?